lib/stages/visTranspose.cpp

```diff
@@ -247,6 +251,9 @@ void visTranspose::main_thread() {
             frame.fpga_seq_length == 0 ? 0. : float(frame.rfi_total) / frame.fpga_seq_length;
         strided_copy(frame.gain.data(), gain.data(), offset * num_input + ti, write_t, num_input);
 
+        // Store original dataset ID (before adding time axis)
+        dset_id[offset + ti] = base_ds_id;
```

Comment:
but why?

Reply:
I don't think I know what you mean...

Reply:
Why do we store the ID of the **base** dataset?

Reply:
the `timeState` was added by the reader just as a convenience to communicate the time axis. It isn't really part of the dataset state that describes the data, which can now potentially be changing in time.